base/dict.jl

```diff
@@ -59,21 +59,77 @@ Given a single iterable argument, constructs a [`Dict`](@ref) whose key-value pa
 are taken from 2-tuples `(key,value)` generated by the argument.
 
 # Examples
-```jldoctest
-julia> Dict([("A", 1), ("B", 2)])
+```jldoctest dictexamples
+julia> d = Dict([("A", 1), ("B", 2)])
 Dict{String,Int64} with 2 entries:
   "B" => 2
   "A" => 1
 ```
 
 Alternatively, a sequence of pair arguments may be passed.
 
-```jldoctest
-julia> Dict("A"=>1, "B"=>2)
+```jldoctest dictexamples
+julia> d = Dict("A"=>1, "B"=>2)
 Dict{String,Int64} with 2 entries:
   "B" => 2
   "A" => 1
 ```
+
+After creation of a Dict you can use `d[x]` for getting the values and `d[x]=y` for changing the values.
+```jldoctest dictexamples
+julia> d["A"]
+1
+
+julia> d["B"] = 3
+3
+```
+
+# Extended help
+
+You can use [Generator](https://docs.julialang.org/en/v1/manual/arrays/#Generator-Expressions-1) and [Comprehension][https://docs.julialang.org/en/v1/manual/arrays/#Comprehensions-1] syntax to create dictionaries:
+
+```jldoctest
+julia> Dict(x => x^2 for x = 0:1:4)
+Dict{Int64,Int64} with 5 entries:
+  0 => 0
+  4 => 16
+  2 => 4
+  3 => 9
+  1 => 1
+```
+
+You can also create and fill dictionaries dynamically (will be slower than the other ways).
+
+First, you need to create an empty dictionary using:
+```jldoctest
+julia> Dict{Int64,Any}()  # type specified
+Dict{Int64,Any} with 0 entries
+
+julia> Dict()  # type not specified
+Dict{Any,Any} with 0 entries
+```
+
+After creating an empty dictionary, you can fill it via a for loop:
+```jldoctest dictexamples
+julia> animaldict = Dict();
+
+julia> animals = ["cats", "fishes", "elephants"];
+
+julia> for (i, x) in enumerate(animals)
+   animaldict[x] = [i, rand(UInt8)]
```

Comment:
You have to chose something deterministic as a value. Otherwise your doctest will output something different each time it is run, making CI fail.

Reply:
Thanks! Fixed in https://github.com/JuliaLang/julia/pull/33936/commits/0b3049d415fc676d33de672cc2506ca630f0c295